File: gaia-sdk-javascript/src/graphql/request/type/EdgeKeyOne.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * This entity represents the output of an edge delete impulse
 */
export class EdgeKeyOne extends Array<(_:VariableRegistry) => string> {
public _typeName = "EdgeKeyOne";
    public source = () => { 
        this.push(_ => "source")
    };

    public edgeId = () => { 
        this.push(_ => "edgeId")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/EdgeKeyOne.py
from gaia_sdk.api.VariableRegistry import VariableRegistry


class EdgeKeyOne(list):
    """
    This entity represents the output of an edge delete impulse
    """

    def source(self):
        self.append(lambda x: "source")

    def edge_id(self):
        self.append(lambda x: "edgeId")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
